docs/guides/themeable-to-emotion-migration-guide.md

```diff
@@ -0,0 +1,459 @@
+---
+title: Themeable to Emotion Migration Guide for Version 8.0
+category: Guides
+order: 5
+---
+
+## Themeable to Emotion Migration Guide for Version 8.0
+
+The main feature of the Version 8.0 release is the switch of the theming engine. We replaced the in-house `ui-themeable` theming solution with the popular [emotion.js](https://emotion.sh/) library.
+
+We feel that it is very important that InstUI is easy to use and learn, so we try to use popular, well maintained and well documented open source solutions whenever possible.
+
+This is why we decided to replace our custom-made theming library (ui-themeable) with a broadly adopted one. The two main candidates were [emotion](https://emotion.sh) and [styled components](https://styled-components.com), we decided to use `emotion`. If you are interested in our detailed reasoning you can read it [here](https://gist.github.com/serikjensen/4ba00b653efac1dbf80543c529adabbc).
+
+- [The emotion package](#themeable-to-emotion-migration-guide/#themeable-to-emotion-migration-guide-for-version-8.0-the-emotion-package)
+- [ApplyTheme to EmotionThemeProvider](#themeable-to-emotion-migration-guide/#applyTheme-to-emotionThemeProvider)
+- [Component Migration steps](#themeable-to-emotion-migration-guide/#component-migration-steps)
+  - [Refactor theme.js](#themeable-to-emotion-migration-guide/#component-migration-steps-refactor-theme.js)
+  - [Create the styles.js file](#themeable-to-emotion-migration-guide/#component-migration-steps-create-the-styles.js-file)
+  - [Make changes in the component](#themeable-to-emotion-migration-guide/#component-migration-steps-make-changes-in-the-component)
+  - [Example Component](#themeable-to-emotion-migration-guide/#component-migration-steps-example-component)
+  - [Theme tests](#themeable-to-emotion-migration-guide/#component-migration-steps-theme-tests)
```

Comment:
If this a table of contents? If yes you should write it above

Reply:
Or even better split this into 2 parts:
1. The parts that are needed for generally the migration (the part that you need to do if you did not made any custom components using themeable) 
2.  the parts that are needed if you made custom components using themeable

Reply:
Added the ToC title, renamed some titles and added a few lines of explanation: https://github.com/instructure/instructure-ui/pull/416/commits/99a30a9cfa305f91b6711f1626288eb85c0037e3

Reply:
looks good!